timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceTransactionIntegrationTest.java

```diff
@@ -0,0 +1,144 @@
+
+    private static final TableReference TABLE = TableReference.create(Namespace.create("test"), "test");
+    private static final byte[] DATA = "foo".getBytes();
+    private static final Cell CELL = Cell.create("bar".getBytes(), "baz".getBytes());
+
+    private static final LockRequest EXCLUSIVE_ADVISORY_LOCK_REQUEST = LockRequest.builder(
+            ImmutableSortedMap.of(StringLockDescriptor.of("foo"), LockMode.WRITE))
+            .build();
+
+    private final ExecutorService executor = Executors.newCachedThreadPool();
+    private final SerializableTransactionManager txnManager;
+
+    public AsyncTimelockServiceTransactionIntegrationTest(TestableTimelockCluster cluster) {
+        super(cluster);
+
+        List<String> serverUris = cluster.servers().stream()
+                .map(server -> server.serverHolder().getTimelockUri())
+                .collect(Collectors.toList());
+        AtlasDbConfig config = ImmutableAtlasDbConfig.builder()
+                .keyValueService(new InMemoryAtlasDbConfig())
+                .timelock(ImmutableTimeLockClientConfig.builder()
+                        .client(CLIENT)
+                        .serversList(ImmutableServerListConfig.builder()
+                                .servers(serverUris)
+                                .build())
+                        .build())
+                .build();
+        txnManager = TransactionManagers.create(config, () -> Optional.empty(), ImmutableSet.of(),
+                ignored -> { }, false);
+        txnManager.getKeyValueService().createTable(TABLE, AtlasDbConstants.GENERIC_TABLE_METADATA);
+    }
+
+    @Test
+    public void canExecuteWriteTransactions() {
+        // write a value
+        txnManager.runTaskWithRetry(txn -> {
+            txn.put(TABLE, ImmutableMap.of(CELL, DATA));
+            return null;
+        });
+
+        // read the value and write a new one
+        byte[] retrievedData = txnManager.runTaskWithRetry(txn -> {
+            byte[] existing = txn.get(TABLE, ImmutableSet.of(CELL)).get(CELL);
+            txn.put(TABLE, ImmutableMap.of(CELL, DATA));
+            return existing;
+        });
+
+        assertThat(retrievedData).isEqualTo(DATA);
+    }
+
+    @Test
+    public void canUseExclusiveAdvisoryLocks() throws ExecutionException, InterruptedException {
+        AtomicBoolean isExecuting = new AtomicBoolean(false);
+
+        List<Future<?>> tasks = IntStream.range(0, 5).mapToObj(i -> executor.submit((Callable<Void>) () ->
+                txnManager.runTaskWithLocksWithRetry(() -> EXCLUSIVE_ADVISORY_LOCK_REQUEST, (txn, locks) -> {
+                    assertTrue(isExecuting.compareAndSet(false, true));
+                    Uninterruptibles.sleepUninterruptibly(500L, TimeUnit.MILLISECONDS);
+                    assertTrue(isExecuting.compareAndSet(true, false));
+
+                    txn.put(TABLE, ImmutableMap.of(CELL, DATA));
```

Comment:
nit: might be worth a brief comment explaining why this is needed (since you still get the mutual-exclusion assertions even if you don't have this line) - we want to ensure that they work with write transactions. Or maybe add the qualification to the test - `canUseExclusiveAdvisoryLocksForWriteTransactions()`